### PR TITLE
[CU-ht1zvt] Split mainLayout to components

### PR DESCRIPTION
**Pull request by Buglik**
no visual changes, just separating mainLayout to components

### CODE REVIEW

**Comment by Zummek** [comment]
Task linked: [CU-ht1zvt Split mainLayout to components](https://app.clickup.com/t/ht1zvt)

File: web-app/src/components/mainLayout/ToolBar.vue [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<template>
```
Comment (eastemployeer) [inline web-app/src/components/mainLayout/ToolBar.vue]: Isn't toolbar a header? Header or Navbar is much more common name than toolbar.

File: web-app/src/components/mainLayout/AvatarMenu.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
+<template>
```
Comment (Zummek) [inline web-app/src/components/mainLayout/AvatarMenu.vue]: It is not generic component, i suggest to rename to `HeaderMenu`, what do you think?

File: web-app/src/components/mainLayout/AvatarMenu.vue [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  <div class="row col-4 justify-end">
+    <q-btn round class="q-ma-sm">
+      <q-avatar :size="$q.screen.lt.sm ? '40px' : '50px'">
+        <img src="https://cdn.quasar.dev/img/boy-avatar.png" />
+        <!-- TODO user avatar -->
+      </q-avatar>
```
Comment (Zummek) [inline web-app/src/components/mainLayout/AvatarMenu.vue]: this component is already made and ready to use

File: web-app/src/components/mainLayout/UserPanel.vue [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    />
+  </div>
+</template>
+
+<script lang="ts">
+import { Vue, Component } from 'vue-property-decorator';
+import User from 'app/../shared/types/User';
+import store from 'src/store';
+
+@Component({})
```
Comment (Zummek) [inline web-app/src/components/mainLayout/UserPanel.vue]:
```suggestion
@Component
```
It could be easier, but less important

**Review by Zummek — APPROVED** [review]

File: web-app/src/components/mainLayout/AvatarMenu.vue (hunk 2 above)
Comment (Buglik) [reply web-app/src/components/mainLayout/AvatarMenu.vue]: how about `HeaderAvatarMenu` or `HeaderSettingsMenu` ?

File: web-app/src/components/mainLayout/AvatarMenu.vue (hunk 3 above)
Comment (Buglik) [reply web-app/src/components/mainLayout/AvatarMenu.vue]: what do you mean ?

Comment (Buglik) [reply web-app/src/components/mainLayout/AvatarMenu.vue]: nvm found it

File: web-app/src/components/mainLayout/AvatarMenu.vue (hunk 2 above)
Comment (Zummek) [reply web-app/src/components/mainLayout/AvatarMenu.vue]: `HeaderAvatarMenu` not bad ;)

File: web-app/src/components/mainLayout/HeaderAvatarMenu.vue [hunk 5]
```diff
@@ -1,10 +1,10 @@
 <template>
   <div class="row col-4 justify-end">
     <q-btn round class="q-ma-sm">
-      <q-avatar :size="$q.screen.lt.sm ? '40px' : '50px'">
-        <img src="https://cdn.quasar.dev/img/boy-avatar.png" />
-        <!-- TODO user avatar -->
-      </q-avatar>
+      <user-avatar
+        :avatar-props="$q.screen.lt.sm ? { size: '40px' } : { size: '50px' }"
```
Comment (Zummek) [inline web-app/src/components/mainLayout/HeaderAvatarMenu.vue:5]: try:
:avatar-props="{ size: $q.screen.lt.sm ? '40px' : '50px' }"

File: web-app/src/components/mainLayout/HeaderAvatarMenu.vue [hunk 6]
```diff
@@ -1,10 +1,10 @@
...
+        :user="$store.state.currentUser.avatarSrc"
```
Comment (Zummek) [inline web-app/src/components/mainLayout/HeaderAvatarMenu.vue]: It should be User not avatarSrc

**Pull request merged**